R/estimate_uncertainty.R

```diff
@@ -15,19 +15,28 @@
 #'   matrices with as many rows as available given the truncation.
 #' @param n Integer indicating the number of reporting matrices to use to
 #'    estimate the dispersion parameters.
-#' @param fun_to_aggregate Function that will operate along the nowcast
-#'    vectors after summing across delays. Eventually, we can add things like
-#'    mean, but for now since we are only providing a negative binomial
-#'    observation model, we can only allow sum. Currently supported
-#'    functions: `sum`.
-#' @param k Integer indicating the number of reference times to apply the
-#'    `fun_to_aggregate` over to create target used to compute the nowcast
-#'    errors.
+#' @param error_model Function that ingests a matrix of observations and a
+#'     matrix of predictions and returns a vector that can be used to
+#'     apply uncertainty using the same error model. Default is
+#'     fit_distribution.
+#' @param error_args List of arguments needed for the specified error model.
+#'     Default is `list(observation_model_name = "negative_binomial").`
+#' @param ref_time_aggregator Function that operates along the rows (reference
```

Comment:
we should need no error args here pass the full function. The default should be a passthrough function. Then we can drop zoo as a dep